src/Operation/Recommendation/Filters.php

```diff
@@ -0,0 +1,215 @@
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Operation\Recommendation;
+
+class Filters
+{
+    /** @var string[]  */
+    private $brands = [];
+
+    /** @var string[]  */
+    private $categories = [];
+
+    /** @var array  */
+    private $customFields = [];
+
+    /** @var boolean */
+    private $discounted;
+
+    /** @var boolean */
+    private $fresh;
+
+    /** @var array */
+    private $price;
+
+    /** @var string[] */
+    private $productIds;
+
+    /** @var float */
+    private $rating;
+
+    /** @var int */
+    private $reviews;
+
+    /** @var string */
+    private $search;
+
+    private $stock;
+    private $tag1;
+    private $tag2;
+    private $tag3;
+
+    /**
+     * @param string[] $brands
+     */
+    public function setBrands(array $brands)
+    {
+        $this->brands = array_merge($this->brands, $brands);
+    }
+
+    /**
+     * @param string[] $categories
+     */
+    public function setCategories(array $categories)
+    {
+        $this->categories = array_merge($this->categories, $categories);
+    }
+
+    /**
+     * @param string $attribute
+     * @param array $values
+     */
+    public function setCustomFields($attribute, array $values)
+    {
+        foreach ($this->customFields as $customField) {
+            if ($customField['attribute'] === $attribute) {
+                $customField['values'] = array_merge($customField['values'], $values);
+            }
+        }
+
+        $this->customFields[] = [
+            'attribute' => $attribute,
+            'values' => $values
+        ];
+    }
+
+    /**
+     * @param boolean $discounted
+     */
+    public function setDiscounted($discounted)
+    {
+        $this->discounted = $discounted;
+    }
+
+    /**
+     * @param boolean $fresh
+     */
+    public function setFresh($fresh)
```

Comment:
`isFresh`?

Reply:
The graphql query field is named `fresh`, just sticked to that.